src/basics.md

```diff
@@ -1244,6 +1245,60 @@ fn main() {
 }
 ```
 
+[ex-error-chain-simple-error-handling]: #ex-error-chain-simple-error-handling
+<a name="ex-error-chain-simple-error-handling"></a>
+## Handle errors correctly in main
+
+[![error-chain-badge]][error-chain] [![cat-rust-patterns-badge]][cat-rust-patterns]
+
+Handles error that occur when trying to open a file that does not
+exist. It is achieved by using [error-chain], a library that takes
+care of a lot of boilerplate code needed in order to [handle errors in
+Rust].
+
+`Io(std::io::Error)` inside [`foreign_links`] allows automatic
+conversion from [`std::io::Error`] into [`error_chain!`] defined type
+implementing the [`Error`] trait.
+
+The below recipe will tell how long the system has been running by
+opening the Unix file `/proc/uptime` and parse the content to get the
+first number. `read_uptime` returns a `Result`, which is the type that
```

Comment:
These last two sentences repeat themselves.  Can we remove the sentence "`read_uptime` returns a `Result`, which is the type that represents a possibility of failure."   We're ready to merge otherwise!